.github/workflows/run-tests.yml

```diff
@@ -1,19 +1,64 @@
 name: Run-Tests
+
 on:
   push:
     branches: [ main ]
   pull_request:
     branches: [ main ]
 
 jobs:
```

Comment:
Your casing for job names is inconsistent. It looks like you predominately use lowercase only, so I'd roll with that.